Automattic/wp-calypso
# Tooling: Add volta versions

#### Changes proposed in this Pull Request

https://volta.sh/

This prepares the environment for use with Volta, a potential replacement for `nvm` (but more).

It sets a `volta` property in the root package.json. It also points _all the other package.json_ in the repo to extend that root config. This is under the assumption that all tooling should use the version in the root.

It was a little annoying to add a volta property to _every_ package.json, but that should be a one-time thing and doesn't seem like it should have any negative impact. That seems to be the way to do things in Volta at the moment. See https://github.com/volta-cli/volta/issues/906 and the [Workspaces docs](https://docs.volta.sh/advanced/workspaces).

#### Testing instructions

You'll need to set your system up for Volta. I have no idea if that will play well with nvm. I've removed nvm from my system completely.

From an arbitrary location like `~`, install some volta tools (that don't correspond to the versions in Calypso). For example:

- `volta install node@latest`
- `volta install typescript@beta`

Check the versions with `volta list`:

```none
$ volta list
⚡️ Currently active tools:

    Node: v15.6.0 (default)
    npm: v6.14.11 (default)
    Yarn: v1.22.10 (default)
    Tool binaries available:
        wp-env (default)
        neovim-node-host (default)
        tsc, tsserver (default)
        yarn-deduplicate (default)

$ node --version
v15.6.0
$ tsc --version
Version 4.2.0-dev
```

move to Calypso and check the versions:

```
$ volta list
⚡️ Currently active tools:

    Node: v12.20.1 (current @ /Users/jonsurrell/a8c/calypso/package.json)
    Yarn: v1.22.10 (current @ /Users/jonsurrell/a8c/calypso/package.json)
    Tool binaries available:
        wp-env (default)
        neovim-node-host (default)
        tsc, tsserver (current @ /Users/jonsurrell/a8c/calypso/package.json)
        yarn-deduplicate (default)

# Nice, volta picked up different versions! Let's check…

$ node --version
# installs on demand if necessary!!
v12.20.1

# It detects the package's version and prefers it!
$ tsc --version
Version 4.1.3
```

Repeat from another directory with a package.json such as `packages/*/` or `apps/editing-toolkit` or `client`.
